src/platform/api/pythonApi.ts

```diff
@@ -379,7 +379,22 @@ export class InterpreterService implements IInterpreterService {
         }
         return this.interpreterListCachePromise;
     }
-
+    private _waitForAllInterpretersToLoad?: Promise<void>;
+    public async waitForAllInterpretersToLoad(): Promise<void> {
+        if (!this._waitForAllInterpretersToLoad) {
+            const lazyLoadControllers = this.workspace
+                .getConfiguration('jupyter', undefined)
+                .get<boolean>('lazyLoadControllers', false);
+            if (lazyLoadControllers) {
+                return;
+            }
+            this._waitForAllInterpretersToLoad = (async () => {
+                await this.refreshInterpreters();
+                await this.getInterpreters();
+            })();
+        }
+        return this._waitForAllInterpretersToLoad;
```

Comment:
I'm going to dig a bit more, but I'm not sure this promise would ensure that the finders have actually recalculated their kernels when this promise is done. The gap I'm trying to consider is if the controllerLoader is going to move forward while local kernel finder is still recalculating its cache.